src/Lean/Server/CodeActions/UnknownIdentifier.lean

```diff
@@ -149,11 +149,8 @@ def computeDotIdQuery?
   let some expectedType := expectedType?
     | return none
   let typeNames? : Option (Array Name) ← ctx.runMetaM lctx do
-    let resultTypeFn := (← instantiateMVars expectedType).cleanupAnnotations.getAppFn.cleanupAnnotations
-    let .const .. := resultTypeFn
-      | return none
```

Comment:
I assume this branch couldn't happen in the first place? (i.e. if there's a `dotId` CompletionInfo, this is always a constant?)

Reply:
I'll make it return `none` if the array is empty.